### PR TITLE
Updated the request header sent to the OIDC endpoint so it doesn't result in a redirect response when an invalid system access token is provided.

### LINKED ISSUE #46130: Update the request header sent to the OIDC endpoint so it doesn't result in a redirect response when an invalid system access token is provided.
See https://github.com/Azure/azure-sdk-for-cpp/pull/6019 for reference

- Add X-TFS-FedAuthRedirect: Suppress in the request header to the request sent to the OIDC endpoint
- Update TSG and test

### PR DESCRIPTION
**Pull request by ahsonkhan**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/46130

### CODE REVIEW

File: sdk/identity/Azure.Identity/tests/AzurePipelinesCredentialLiveTests.cs [hunk 1]
```diff
@@ -44,5 +44,36 @@ public async Task AzurePipelineCredentialLiveTest_GetToken()
...
+                }
+                Console.WriteLine(sb);
+                Assert.Fail($"{sb} SYSTEM_ACCESSTOKEN: {systemAccessToken}, AZURE_SERVICE_CONNECTION_TENANT_ID: {tenantId}, AZURE_SERVICE_CONNECTION_CLIENT_ID: {clientId}, AZURE_SERVICE_CONNECTION_ID: {serviceConnectionId}");
+                Assert.Ignore("AzurePipelinesCredentialLiveTests disabled because required environment variables are not set");
+            }
+
+            var cred = new AzurePipelinesCredential(tenantId, clientId, serviceConnectionId, systemAccessToken);
+
+            var ex = Assert.ThrowsAsync<AuthenticationFailedException>(async () => await cred.GetTokenAsync(new TokenRequestContext(new[] { "https://management.azure.com//.default" }), CancellationToken.None));
+            Assert.That(ex.Message, Does.Contain("401 (Unauthorized)"));
```
Comment (christothes) [inline sdk/identity/Azure.Identity/tests/AzurePipelinesCredentialLiveTests.cs:73]: nit: We could test the actual status code of the response also.

**Review by christothes — APPROVED** [review]

File: sdk/identity/Azure.Identity/tests/AzurePipelinesCredentialLiveTests.cs (hunk 1 above)
Comment (ahsonkhan) [reply sdk/identity/Azure.Identity/tests/AzurePipelinesCredentialLiveTests.cs:73]: How do I fetch the response containing the status code? It seems like `ErrorResponse` has the fields.
Is there a way for me to save the raw response from the GetToken() call even if it fails with an `AuthenticationFailedException`, and then use that to assert 401?

Comment (ahsonkhan) [reply sdk/identity/Azure.Identity/tests/AzurePipelinesCredentialLiveTests.cs:73]: This is the output from the `GetTokenAsync()` call that I have to work with, which includes the log and exception message output.
If you can provide me with an example of how we test the actual status code, elsewhere in the SDK, I could try to use that pattern. Without mocking, I wasn't able to find an example from the existing Identity live/unit tests.

```text
2024-09-24T00:21:12.7439310Z   Failed AzurePipelineCredentialLiveTest_GetToken_InvalidSystemAccessToken [34 ms]
2024-09-24T00:21:12.7441050Z   Error Message:
2024-09-24T00:21:12.7444160Z      Expected: String containing "401 (Unauthorized)"
2024-09-24T00:21:12.7453120Z   But was:  "OIDC token not found in response. See the troubleshooting guide for more information. https://aka.ms/azsdk/net/identity/azurepipelinescredential/troubleshoot
2024-09-24T00:21:12.7466220Z 
2024-09-24T00:21:12.7482280Z Response= {"$id":"1","innerException":null,"message":"TF400813: The user 'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa' is not authorized to access this resource.","typeName":"Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException, Microsoft.TeamFoundation.Framework.Server","typeKey":"UnauthorizedRequestException","errorCode":0,"eventId":3000}"
2024-09-24T00:21:12.7484130Z 
2024-09-24T00:21:12.7486440Z   Stack Trace:
2024-09-24T00:21:12.7496090Z      at Azure.Identity.Tests.AzurePipelinesCredentialLiveTests.AzurePipelineCredentialLiveTest_GetToken_InvalidSystemAccessToken() in /Users/runner/work/1/s/sdk/identity/Azure.Identity/tests/AzurePipelinesCredentialLiveTests.cs:line 73
2024-09-24T00:21:12.7511980Z    at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter`1.GetResult()
2024-09-24T00:21:12.7525880Z    at NUnit.Framework.Internal.AsyncToSyncAdapter.Await(Func`1 invoke)
2024-09-24T00:21:12.7540880Z    at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
2024-09-24T00:21:12.7551560Z    at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
2024-09-24T00:21:12.7566860Z    at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.<>c__DisplayClass1_0.<Execute>b__0()
2024-09-24T00:21:12.7575200Z    at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.RunTestMethodInThreadAbortSafeZone(TestExecutionContext context, Action action)
2024-09-24T00:21:12.7588450Z 
2024-09-24T00:21:12.7608940Z 1)    at Azure.Identity.Tests.AzurePipelinesCredentialLiveTests.AzurePipelineCredentialLiveTest_GetToken_InvalidSystemAccessToken() in /Users/runner/work/1/s/sdk/identity/Azure.Identity/tests/AzurePipelinesCredentialLiveTests.cs:line 73
2024-09-24T00:21:12.7621910Z    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
2024-09-24T00:21:12.7635950Z    at Azure.Identity.Tests.AzurePipelinesCredentialLiveTests.AzurePipelineCredentialLiveTest_GetToken_InvalidSystemAccessToken()
2024-09-24T00:21:12.7646960Z 
2024-09-24T00:21:12.7659220Z 
2024-09-24T00:21:12.7672420Z   Standard Output Messages:
2024-09-24T00:21:12.7687660Z  [00:21:12.567] GetToken 
2024-09-24T00:21:12.7695780Z    method: AzurePipelinesCredential.GetToken
2024-09-24T00:21:12.7710100Z    scopes: [ https://management.azure.com//.default ]
2024-09-24T00:21:12.7724020Z    parentRequestId: 
2024-09-24T00:21:12.7730430Z  
2024-09-24T00:21:12.7732020Z  
2024-09-24T00:21:12.7734590Z  [00:21:12.567] LogMsalVerbose 
2024-09-24T00:21:12.7737050Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z] ConfidentialClientApplication 4948421 created
2024-09-24T00:21:12.7738940Z  
2024-09-24T00:21:12.7755380Z  
2024-09-24T00:21:12.7767220Z  [00:21:12.567] LogMsalInformational 
2024-09-24T00:21:12.7792860Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] MSAL MSAL.NetCore with assembly version '4.62.0.0'. CorrelationId(14a8fd2f-0726-43a5-8d77-e77d7870dc86)
2024-09-24T00:21:12.7807870Z  
2024-09-24T00:21:12.7820100Z  
2024-09-24T00:21:12.7828600Z  [00:21:12.567] LogMsalInformational 
2024-09-24T00:21:12.7836470Z    message: 
2024-09-24T00:21:12.7844720Z      False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] === AcquireTokenForClientParameters ===
2024-09-24T00:21:12.7846800Z      SendX5C: False
2024-09-24T00:21:12.7849450Z      ForceRefresh: False
2024-09-24T00:21:12.7851340Z  
2024-09-24T00:21:12.7854710Z  
2024-09-24T00:21:12.7856360Z  [00:21:12.567] LogMsalInformational 
2024-09-24T00:21:12.7859020Z    message: 
2024-09-24T00:21:12.7862850Z      False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] 
2024-09-24T00:21:12.7865820Z      === Request Data ===
2024-09-24T00:21:12.7882440Z      Authority Provided? - True
2024-09-24T00:21:12.7903450Z      Scopes - https://management.azure.com//.default
2024-09-24T00:21:12.7924000Z      Extra Query Params Keys (space separated) - 
2024-09-24T00:21:12.7949370Z      ApiId - AcquireTokenForClient
2024-09-24T00:21:12.7969310Z      IsConfidentialClient - True
2024-09-24T00:21:12.7987140Z      SendX5C - False
2024-09-24T00:21:12.8000430Z      LoginHint ? False
2024-09-24T00:21:12.8010050Z      IsBrokerConfigured - False
2024-09-24T00:21:12.8017720Z      HomeAccountId - False
2024-09-24T00:21:12.8032130Z      CorrelationId - 14a8fd2f-0726-43a5-8d77-e77d7870dc86
2024-09-24T00:21:12.8040400Z      UserAssertion set: False
2024-09-24T00:21:12.8048530Z      LongRunningOboCacheKey set: False
2024-09-24T00:21:12.8056240Z      Region configured: 
2024-09-24T00:21:12.8057820Z  
2024-09-24T00:21:12.8060250Z  
2024-09-24T00:21:12.8062720Z  [00:21:12.567] LogMsalInformational 
2024-09-24T00:21:12.8071570Z    message: 
2024-09-24T00:21:12.8084560Z      False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] === Token Acquisition (ClientCredentialRequest) started:
2024-09-24T00:21:12.8098780Z      	 Scopes: https://management.azure.com//.default
2024-09-24T00:21:12.8108370Z      	Authority Host: ***
2024-09-24T00:21:12.8127610Z  
2024-09-24T00:21:12.8141340Z  
2024-09-24T00:21:12.8150600Z  [00:21:12.567] LogMsalAlways 
2024-09-24T00:21:12.8164340Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] [Internal cache] Total number of cache partitions found while getting access tokens: 0
2024-09-24T00:21:12.8179010Z  
2024-09-24T00:21:12.8189660Z  
2024-09-24T00:21:12.8200670Z  [00:21:12.568] LogMsalAlways 
2024-09-24T00:21:12.8216500Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] [FindAccessTokenAsync] Discovered 0 access tokens in cache using partition key: ***_***_AppTokenCache
2024-09-24T00:21:12.8224080Z  
2024-09-24T00:21:12.8239170Z  
2024-09-24T00:21:12.8250110Z  [00:21:12.568] LogMsalVerbose 
2024-09-24T00:21:12.8261440Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] [FindAccessTokenAsync] No access tokens found in the cache. Skipping filtering. 
2024-09-24T00:21:12.8276410Z  
2024-09-24T00:21:12.8284740Z  
2024-09-24T00:21:12.8306820Z  [00:21:12.568] LogMsalInformational 
2024-09-24T00:21:12.8321480Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] [Instance Discovery] Instance discovery is enabled and will be performed
2024-09-24T00:21:12.8333140Z  
2024-09-24T00:21:12.8342480Z  
2024-09-24T00:21:12.8356180Z  [00:21:12.568] LogMsalVerbose 
2024-09-24T00:21:12.8364930Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] [Region discovery] WithAzureRegion not configured. 
2024-09-24T00:21:12.8378940Z  
2024-09-24T00:21:12.8390290Z  
2024-09-24T00:21:12.8401920Z  [00:21:12.568] LogMsalInformational 
2024-09-24T00:21:12.8416220Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] [Region discovery] Not using a regional authority. 
2024-09-24T00:21:12.8423540Z  
2024-09-24T00:21:12.8431260Z  
2024-09-24T00:21:12.8438380Z  [00:21:12.568] LogMsalVerbose 
2024-09-24T00:21:12.8446970Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] [Instance Discovery] Tried to use network cache provider for ***. Success? True. 
2024-09-24T00:21:12.8449100Z  
2024-09-24T00:21:12.8451500Z  
2024-09-24T00:21:12.8454500Z  [00:21:12.568] LogMsalVerbose 
2024-09-24T00:21:12.8457090Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] [Instance Discovery] The network provider found an entry for ***. 
2024-09-24T00:21:12.8459540Z  
2024-09-24T00:21:12.8461970Z  
2024-09-24T00:21:12.8464720Z  [00:21:12.568] LogMsalVerbose 
2024-09-24T00:21:12.8467100Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] Starting TokenClient:SendTokenRequestAsync
2024-09-24T00:21:12.8469090Z  
2024-09-24T00:21:12.8471470Z  
2024-09-24T00:21:12.8473840Z  [00:21:12.568] LogMsalVerbose 
2024-09-24T00:21:12.8476860Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] [TokenClient] Before adding the client assertion / secret
2024-09-24T00:21:12.8478930Z  
2024-09-24T00:21:12.8481440Z  
2024-09-24T00:21:12.8483220Z  [00:21:12.568] Request f03b52a9-c34c-43f5-a909-ea506240bed3
2024-09-24T00:21:12.8484980Z    method: POST
2024-09-24T00:21:12.8487370Z    uri: https://dev.azure.com/azure-sdk/590cfd2a-581c-4dcb-a12e-6568ce786175/_apis/distributedtask/hubs/build/plans/2e011469-bc90-4dc7-9242-012be179cb56/jobs/5ee5fa9b-185c-52bc-0624-8e87d1d5c9f8/oidctoken?api-version=7.1&serviceConnectionId=REDACTED
2024-09-24T00:21:12.8489920Z    headers: 
2024-09-24T00:21:12.8492470Z      Authorization:REDACTED
2024-09-24T00:21:12.8494210Z      Content-Type:application/json
2024-09-24T00:21:12.8495980Z      X-TFS-FedAuthRedirect:REDACTED
2024-09-24T00:21:12.8497850Z      x-ms-client-request-id:f03b52a9-c34c-43f5-a909-ea506240bed3
2024-09-24T00:21:12.8507420Z      x-ms-return-client-request-id:true
2024-09-24T00:21:12.8510010Z      User-Agent:azsdk-net-Identity/1.13.0-alpha.20240923.2 (.NET 6.0.33; Darwin 21.6.0 Darwin Kernel Version 21.6.0: Mon Jun 24 00:56:10 PDT 2024; root:xnu-8020.240.18.709.2~1/RELEASE_X86_64)
2024-09-24T00:21:12.8512070Z    clientAssembly: Azure.Identity
2024-09-24T00:21:12.8513530Z  
2024-09-24T00:21:12.8514980Z  
2024-09-24T00:21:12.8516660Z  [00:21:12.599] ErrorResponse f03b52a9-c34c-43f5-a909-ea506240bed3
2024-09-24T00:21:12.8518350Z    status: 401
2024-09-24T00:21:12.8520930Z    reasonPhrase: Unauthorized
2024-09-24T00:21:12.8522480Z    headers: 
2024-09-24T00:21:12.8524110Z      Cache-Control:no-cache
2024-09-24T00:21:12.8525710Z      Pragma:no-cache
2024-09-24T00:21:12.8528400Z      P3P:REDACTED
2024-09-24T00:21:12.8530820Z      WWW-Authenticate:Bearer authorization_uri=***/***,Basic realm="https://tfsprodcus3.visualstudio.com/",TFS-Federated
2024-09-24T00:21:12.8532810Z      X-TFS-ProcessId:REDACTED
2024-09-24T00:21:12.8535770Z      Strict-Transport-Security:REDACTED
2024-09-24T00:21:12.8537490Z      ActivityId:REDACTED
2024-09-24T00:21:12.8540140Z      X-TFS-Session:REDACTED
2024-09-24T00:21:12.8541770Z      X-VSS-E2EID:REDACTED
2024-09-24T00:21:12.8544440Z      X-VSS-SenderDeploymentId:REDACTED
2024-09-24T00:21:12.8546170Z      X-Frame-Options:REDACTED
2024-09-24T00:21:12.8548900Z      X-TFS-FedAuthRealm:REDACTED
2024-09-24T00:21:12.8550660Z      X-TFS-FedAuthIssuer:REDACTED
2024-09-24T00:21:12.8552380Z      X-VSS-AuthorizationEndpoint:REDACTED
2024-09-24T00:21:12.8554760Z      X-VSS-ResourceTenant:REDACTED
2024-09-24T00:21:12.8556620Z      X-VSS-S2STargetService:REDACTED
2024-09-24T00:21:12.8558350Z      X-TFS-FedAuthRedirect:REDACTED
2024-09-24T00:21:12.8560630Z      Request-Context:REDACTED
2024-09-24T00:21:12.8562380Z      Access-Control-Expose-Headers:REDACTED
2024-09-24T00:21:12.8564780Z      X-Content-Type-Options:REDACTED
2024-09-24T00:21:12.8566500Z      X-Cache:REDACTED
2024-09-24T00:21:12.8569100Z      X-MSEdge-Ref:REDACTED
2024-09-24T00:21:12.8571330Z      Date:Tue, 24 Sep 2024 00:21:12 GMT
2024-09-24T00:21:12.8573640Z      Content-Length:343
2024-09-24T00:21:12.8575370Z      Content-Type:application/json; charset=utf-8
2024-09-24T00:21:12.8578230Z      Expires:-1
2024-09-24T00:21:12.8579860Z    seconds: 0.031690698
2024-09-24T00:21:12.8581330Z  
2024-09-24T00:21:12.8582800Z  
2024-09-24T00:21:12.8584240Z  [00:21:12.6] LogMsalVerbose 
2024-09-24T00:21:12.8586970Z    message: False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] Finished TokenClient:SendTokenRequestAsync in 32 ms
2024-09-24T00:21:12.8589460Z  
2024-09-24T00:21:12.8592140Z  
2024-09-24T00:21:12.8593760Z  [00:21:12.601] LogMsalError 
2024-09-24T00:21:12.8595270Z    message: 
2024-09-24T00:21:12.8598590Z      False MSAL 4.62.0.0 MSAL.NetCore .NET 6.0.33 MacOS [2024-09-24 00:21:12Z - 14a8fd2f-0726-43a5-8d77-e77d7870dc86] Exception type: Azure.Identity.AuthenticationFailedException
2024-09-24T00:21:12.8601140Z      To see full exception details, enable PII Logging. See https://aka.ms/msal-net-logging
2024-09-24T00:21:12.8603390Z         at Azure.Identity.AzurePipelinesCredential.GetOidcTokenResponse(HttpMessage message) in /_/sdk/identity/Azure.Identity/src/Credentials/AzurePipelinesCredential.cs:line 156
2024-09-24T00:21:12.8606020Z         at Azure.Identity.AzurePipelinesCredential.<>c__DisplayClass22_0.<<-ctor>b__0>d.MoveNext() in /_/sdk/identity/Azure.Identity/src/Credentials/AzurePipelinesCredential.cs:line 62
2024-09-24T00:21:12.8608170Z      --- End of stack trace from previous location ---
2024-09-24T00:21:12.8611690Z         at Microsoft.Identity.Client.Internal.ClientCredential.SignedAssertionDelegateClientCredential.AddConfidentialClientParametersAsync(OAuth2Client oAuth2Client, ILoggerAdapter logger, ICryptographyManager cryptographyManager, String clientId, String tokenEndpoint, Boolean sendX5C, Boolean useSha2, CancellationToken cancellationToken)
2024-09-24T00:21:12.8614720Z         at Microsoft.Identity.Client.OAuth2.TokenClient.AddBodyParamsAndHeadersAsync(IDictionary`2 additionalBodyParameters, String scopes, CancellationToken cancellationToken)
2024-09-24T00:21:12.8617340Z         at Microsoft.Identity.Client.OAuth2.TokenClient.SendTokenRequestAsync(IDictionary`2 additionalBodyParameters, String scopeOverride, String tokenEndpointOverride, CancellationToken cancellationToken)
2024-09-24T00:21:12.8619930Z         at Microsoft.Identity.Client.Internal.Requests.RequestBase.SendTokenRequestAsync(IDictionary`2 additionalBodyParameters, CancellationToken cancellationToken)
2024-09-24T00:21:12.8622340Z         at Microsoft.Identity.Client.Internal.Requests.ClientCredentialRequest.GetAccessTokenAsync(CancellationToken cancellationToken, ILoggerAdapter logger)
2024-09-24T00:21:12.8634360Z         at Microsoft.Identity.Client.Internal.Requests.ClientCredentialRequest.ExecuteAsync(CancellationToken cancellationToken)
2024-09-24T00:21:12.8636960Z         at Microsoft.Identity.Client.Internal.Requests.RequestBase.<>c__DisplayClass11_1.<<RunAsync>b__1>d.MoveNext()
2024-09-24T00:21:12.8642910Z      --- End of stack trace from previous location ---
2024-09-24T00:21:12.8648770Z         at Microsoft.Identity.Client.Utils.StopwatchService.MeasureCodeBlockAsync(Func`1 codeBlock)
2024-09-24T00:21:12.8654160Z         at Microsoft.Identity.Client.Internal.Requests.RequestBase.RunAsync(CancellationToken cancellationToken)
2024-09-24T00:21:12.8660440Z  
2024-09-24T00:21:12.8662370Z  
2024-09-24T00:21:12.8665550Z  [00:21:12.601] GetTokenFailed 
2024-09-24T00:21:12.8668170Z    method: AzurePipelinesCredential.GetToken
2024-09-24T00:21:12.8670510Z    scopes: [ https://management.azure.com//.default ]
2024-09-24T00:21:12.8673040Z    parentRequestId: 
2024-09-24T00:21:12.8675250Z    exception: 
2024-09-24T00:21:12.8677880Z      Azure.Identity.AuthenticationFailedException (0x80131500): OIDC token not found in response. See the troubleshooting guide for more information. https://aka.ms/azsdk/net/identity/azurepipelinescredential/troubleshoot
2024-09-24T00:21:12.8681330Z      Response= {"$id":"1","innerException":null,"message":"TF400813: The user 'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa' is not authorized to access this resource.","typeName":"Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException, Microsoft.TeamFoundation.Framework.Server","typeKey":"UnauthorizedRequestException","errorCode":0,"eventId":3000}
```

Comment (christothes) [reply sdk/identity/Azure.Identity/tests/AzurePipelinesCredentialLiveTests.cs]: Sorry, I was thinking we had access to the response also - but it seems we don't.

**Comment by ahsonkhan** [comment]
@christothes I keep seeing a seemingly unrelated test fail on [Ubuntu](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4163699&view=logs&j=e08f9da5-dedf-508d-cb81-dc53e622c73b&t=96ffab2b-fe3c-5ce1-8cb5-7687a41af6c0):
`Failed KubectlExecuteIdentityAKSTests`
It fails on retry.

Is this a known issue? It fails on identity test runs outside this PR, which makes me believe it isn't related to it.
Any recommendations on how to unblock this PR? I have gotten enough confidence of correctness from the passing tests (and new test addition) in the `net - identity - tests` legs, so I could reset and run just the CI pipelines/unit tests. Does that work?

```text
2024-09-24T20:33:20.0037474Z   Failed KubectlExecuteIdentityAKSTests [754 ms]
2024-09-24T20:33:20.0038481Z   Error Message:
2024-09-24T20:33:20.0041045Z    System.InvalidOperationException: ERROR: AADSTS700024: Client assertion is not within its valid time range. Current time: 2024-09-24T20:33:19.6123783Z, assertion valid from 2024-09-24T20:16:20.0000000Z, expiry time of assertion 2024-09-24T20:26:20.0000000Z. Review the documentation at https://learn.microsoft.com/entra/identity-platform/certificate-credentials . Trace ID: 4edaea99-30f4-4c2f-83f5-a7b4e83f3000 Correlation ID: 951d6530-5f14-4a20-a9a5-aba45e411644 Timestamp: 2024-09-24 20:33:19Z
2024-09-24T20:33:20.0044185Z Interactive authentication is needed. Please run:
2024-09-24T20:33:20.0046014Z az login
2024-09-24T20:33:20.0048084Z    at Azure.Identity.ProcessRunner.Run() in /_/sdk/identity/Azure.Identity/src/ProcessRunner.cs:line 66
2024-09-24T20:33:20.0050003Z    at Azure.Identity.Tests.ManagedIdentityAKSIntegrationTests.RunCommand(String fileName, String args) in /mnt/vss/_work/1/s/sdk/identity/Azure.Identity/tests/ManagedIdentityAKSIntegrationTests.cs:line 76
2024-09-24T20:33:20.0052414Z   Stack Trace:
2024-09-24T20:33:20.0054820Z      at Azure.Identity.Tests.ManagedIdentityAKSIntegrationTests.RunCommand(String fileName, String args) in /mnt/vss/_work/1/s/sdk/identity/Azure.Identity/tests/ManagedIdentityAKSIntegrationTests.cs:line 83
2024-09-24T20:33:20.0058030Z    at Azure.Identity.Tests.ManagedIdentityAKSIntegrationTests.SetupKubernetesEnvironment() in /mnt/vss/_work/1/s/sdk/identity/Azure.Identity/tests/ManagedIdentityAKSIntegrationTests.cs:line 41
2024-09-24T20:33:20.0061136Z    at Azure.Identity.Tests.ManagedIdentityAKSIntegrationTests.KubectlExecuteIdentityAKSTests() in /mnt/vss/_work/1/s/sdk/identity/Azure.Identity/tests/ManagedIdentityAKSIntegrationTests.cs:line 60
2024-09-24T20:33:20.0063491Z 
2024-09-24T20:33:20.0067912Z 1)    at Azure.Identity.Tests.ManagedIdentityAKSIntegrationTests.RunCommand(String fileName, String args) in /mnt/vss/_work/1/s/sdk/identity/Azure.Identity/tests/ManagedIdentityAKSIntegrationTests.cs:line 76
2024-09-24T20:33:20.0070993Z    at Azure.Identity.Tests.ManagedIdentityAKSIntegrationTests.SetupKubernetesEnvironment() in /mnt/vss/_work/1/s/sdk/identity/Azure.Identity/tests/ManagedIdentityAKSIntegrationTests.cs:line 41
2024-09-24T20:33:20.0074273Z    at Azure.Identity.Tests.ManagedIdentityAKSIntegrationTests.KubectlExecuteIdentityAKSTests() in /mnt/vss/_work/1/s/sdk/identity/Azure.Identity/tests/ManagedIdentityAKSIntegrationTests.cs:line 60
2024-09-24T20:33:20.0075791Z 
2024-09-24T20:33:20.0076432Z 
2024-09-24T20:33:20.0078061Z   Standard Output Messages:
2024-09-24T20:33:20.0079856Z  Running command: which az
2024-09-24T20:33:20.0081463Z  output:
2024-09-24T20:33:20.0082969Z  /usr/bin/az
2024-09-24T20:33:20.0084595Z  Running command: which kubectl
2024-09-24T20:33:20.0086169Z  output:
2024-09-24T20:33:20.0087866Z  /usr/bin/kubectl
2024-09-24T20:33:20.0094844Z  Running command: /usr/bin/az login --federated-token *** --service-principal -u *** --tenant ***
2024-09-24T20:33:20.0099150Z  System.InvalidOperationException: ERROR: AADSTS700024: Client assertion is not within its valid time range. Current time: 2024-09-24T20:33:19.6123783Z, assertion valid from 2024-09-24T20:16:20.0000000Z, expiry time of assertion 2024-09-24T20:26:20.0000000Z. Review the documentation at https://learn.microsoft.com/entra/identity-platform/certificate-credentials . Trace ID: 4edaea99-30f4-4c2f-83f5-a7b4e83f3000 Correlation ID: 951d6530-5f14-4a20-a9a5-aba45e411644 Timestamp: 2024-09-24 20:33:19Z
2024-09-24T20:33:20.0102560Z  Interactive authentication is needed. Please run:
2024-09-24T20:33:20.0104500Z  az login
2024-09-24T20:33:20.0106630Z     at Azure.Identity.ProcessRunner.Run() in /_/sdk/identity/Azure.Identity/src/ProcessRunner.cs:line 66
2024-09-24T20:33:20.0110149Z     at Azure.Identity.Tests.ManagedIdentityAKSIntegrationTests.RunCommand(String fileName, String args) in /mnt/vss/_work/1/s/sdk/identity/Azure.Identity/tests/ManagedIdentityAKSIntegrationTests.cs:line 76
2024-09-24T20:33:20.0127239Z 
2024-09-24T20:33:20.0127903Z 
```
From analytics, this particular test has failed intermittently over the last few weeks:
image

**Comment by christothes** [comment]
> @christothes I keep seeing a seemingly unrelated test fail on [Ubuntu](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4163699&view=logs&j=e08f9da5-dedf-508d-cb81-dc53e622c73b&t=96ffab2b-fe3c-5ce1-8cb5-7687a41af6c0): `Failed KubectlExecuteIdentityAKSTests` It fails on retry.
> 
> Is this a known issue? It fails on identity test runs outside this PR, which makes me believe it isn't related to it. Any recommendations on how to unblock this PR? I have gotten enough confidence of correctness from the passing tests (and new test addition) in the `net - identity - tests` legs, so I could reset and run just the CI pipelines/unit tests. Does that work?
> 
> Please let me know if you'd like me to file a tracking issue for it (or optionally disable the test).

Yes, let's file a tracking issue. I think this is just a timing related issue since we are using the OIDC token that was set at the begining of the pipeline and sometimes we just happen to execute those tests after the 10 minute lifetime expiration. I have some ideas for how to resolve it that we can discuss in the issue. 

We can Ignore the test in the meantime, if it is blocking now consistently, and cite the issue in the reason.

**Comment by ahsonkhan** [comment]
> Yes, let's file a tracking issue. 

Done https://github.com/Azure/azure-sdk-for-net/issues/46209

**Review by Thompson1985 — APPROVED** [review]

**Pull request merged**